app/database/queries/breeze_buddy/users.py

```
@@ -51,10 +117,617 @@ async def get_user_by_username(username: str) -> Optional[UserInDB]:
+    query = """
+        SELECT id, username, role, email, merchant_ids, shop_identifiers, is_active, created_at, updated_at
+        FROM users
+        WHERE id = $1 AND role = 'merchant'
+    """
+    try:
+        async for conn in get_db_connection():
+            row = await conn.fetchrow(query, merchant_uuid)
+            if not row:
+                return None
+            return _build_merchant_response(row)
+        raise Exception("No database connection available")
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)